main.cpp

```diff
@@ -213,191 +334,379 @@ struct Computer
+    void changeVolume(bool up); 
+
+    float outputSound(); 
+
+    void changeSource(std::string newSource);
 };
 
+void Amplifier::changeVolume(bool up)
+{
+    float volumeDelta = 0.1f;
+    if (up)
+    {
+        volumeButtonValue += volumeDelta;
+        if (volumeButtonValue > maxVolume)
+        {
+            volumeButtonValue = maxVolume;
+        }
+    }
+    else
+    {
+        volumeButtonValue -= volumeDelta;
+        if (volumeButtonValue < 0)
+        {
+            volumeButtonValue = 0;
+        }
+    }
+}
+
+float Amplifier::outputSound()
+{
+    return static_cast<float>(rand()) / static_cast<float>(RAND_MAX) * volumeButtonValue / maxVolume;
+}
+
+void Amplifier::changeSource(std::string newSource)
+{
+    sourceSelected = newSource;
+}
+
 struct Tuner
 {
-    // band selected
     std::string bandSelected = "FM";
-    // tuning know angle
     float tuningKnobAngleRadians = 1.234f;
-    // preset button pressed
     std::string presetButtonPressed = "none";
-    // dial position
     float dialPositionMillimeters = 15.3f;
-    // antenna signal voltage
     double antennaSignalVoltageMillivolts = 0.132;
+    float tuning;
     
-    // 3 things it can do:
-    // select band
-    bool selectBand(std::string newBand); // returns true if the selection is successful
+    void selectBand(std::string newBand);
 
-    // change tuning
-    float changeTuning(bool up); // returns the new tuning knob rotation angle
+    void changeTuning(bool up); 
 
-    // light up
-    bool lightUp(); // returns true if the light is on
+    void lightUp(); 
 };
 
+void Tuner::selectBand(std::string newBand)
+{
+    bandSelected = newBand;
+}
+
+void Tuner::changeTuning(bool up)
+{   // TODO Tuning and tuning limits for various bands
+    float tuningDelta = 1;
+    if (up)
+    {
+        tuning += tuningDelta;
+    }
+    else
+    {
+        tuning -= tuningDelta;
+    }
+}
+
+void Tuner::lightUp()
+{
+    std::cout << "Dial lit up!" << std::endl;
+}
+
 struct Speaker
 {
-    // bass driver voltage
     double bassDriverVoltageVolts = 31.43;
-    // bass cone displacement
     float bassConeDisplacementMillimeters = 2.3f;
-    // treeble driver voltage
     double treebleDriverVoltageVolts = 30.56;
-    // treeble cone displacement
     float treebleConeDisplacementMillimeters = 0.4f;
-    // vent exhaust pressure
     float ventExhaustOverPressurePa = 45.121f;
+    float maxCurrent = 1.2f;
 
-    // 3 things it can do:
-    // process signal
-    float processSignal(float inputSignal); // returns a driver current 
+    float processSignal(float inputSignal); 
   
-    // emit sound
-    void emitSound(float driverCurrent);
+    float emitSound(float driverCurrent);
     
-    // blow
-    void blow(float driverCurrent);
+    bool blow(float driverCurrent);
 };
 
+float Speaker::processSignal(float inputSignal)
+{
+    return inputSignal * static_cast<float>(rand()) / static_cast<float>(RAND_MAX);
```

Comment:
Casting has not been covered yet.  
Remove this cast.
Change your code so that no warnings are generated **without** using casting.
- [ ] revise